paasta_tools/cli/cmds/status.py

```diff
@@ -1124,6 +1132,112 @@ def print_flink_status(
     return 0
 
 
+def print_kubernetes_status_v2(
+    cluster: str,
+    service: str,
+    instance: str,
+    output: List[str],
+    status,
+    verbose: int = 0,
+) -> int:
+    instance_state = get_instance_state(status)
+    output.append(f"    State: {instance_state}")
+    output.append("    Running versions:")
+    output.append("      " + PaastaColors.green("Rerun with -v to see all replicas"))
+    output.extend([f"      {line}" for line in get_versions_table(status.replicasets)])
+    if status.error_message:
+        output.append("    " + PaastaColors.red(status.error_message))
+        return 1
+    else:
+        return 0
+
+
+def get_instance_state(status) -> str:
+    num_replicasets = len(status.replicasets)
+    num_ready_replicas = sum(r.ready_replicas for r in status.replicasets)
+    if status.desired_state == "stop":
+        if num_replicasets == 0:
+            return PaastaColors.red("Stopped")
+        else:
+            return PaastaColors.red("Stopping")
+    elif status.desired_state == "start":
+        if num_replicasets == 0:
+            return PaastaColors.yellow("Starting")
+        if num_replicasets == 1:
+            if num_ready_replicas < status.desired_instances:
+                return PaastaColors.yellow("Launching replicas")
+            else:
+                return PaastaColors.green("Running")
+        else:
+            replicasets = sorted(status.replicasets, key=lambda x: x.create_timestamp)
+            git_shas = {r.git_sha for r in replicasets}
+            config_shas = {r.config_sha for r in replicasets}
+            bouncing_to = []
+            if len(git_shas) > 1:
+                bouncing_to.append(replicasets[0].git_sha[:8])
+            if len(config_shas) > 1:
+                bouncing_to.append(replicasets[0].config_sha)
+
+            bouncing_to_str = ", ".join(bouncing_to)
+            return PaastaColors.yellow(f"Bouncing to {bouncing_to_str}")
```

Comment:
The bouncing state could be: `Bouncing to abc123, configdef456`, which could be confusing. The git SHA on it's own would be intuitive, but the on it's own or with the git SHA seems confusing. Could we explain this more or not show it somehow? Maybe say "due to a SOA configs change"? Hmm, but it could due to a big bounce too...

Reply:
yeah I am not sure.  I guess I would like to leave this as-is for now and we can revisit it